[Web] Fix handling of `enabled` prop

m-bert opened the pull request:
## Description

Right now, if one changes `enabled` property of handlers to `false`, they can never activate again. In the following example:

```ts
const [enabled, setEnabled] = useState(false);
const pan = Gesture.Pan().enabled(enabled)
```

gesture will never become active, even if `enabled` changes to `true`. Same happens when `enabled` is set to `true` and then changed to `false` and `true` again. 

This happens because we set `enabled` to `true` by default in `setGestureConfig`, thus the following check:

```ts
if (config.enabled !== undefined && this.enabled !== config.enabled) { ... }
```
 passes only when `config.enabled` is `false`.

I've set `enabled` to `null` by default, and then if `config.enabled` is defined, it is assigned given value. Else default will be `true`. To fix the issue, I've removed reset of `enabled` in `resetConfig` method.

Alternative approach would be to follow Android implementation and always react to changes in `enabled` even in cases like `false` $\rightarrow$ `false` and `true` $\rightarrow$ `true`.

## Test plan

<details>
<summary>Tested on the following code:</summary>

```tsx
import React from 'react';
import { Button, StyleSheet, View } from 'react-native';
import {
  Gesture,
  GestureDetector,
  GestureHandlerRootView,
  NativeDetector,
  usePan,
} from 'react-native-gesture-handler';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
} from 'react-native-reanimated';

export default function EmptyExample() {
  const [showDetector, setShowDetector] = React.useState(true);

  const [enablePan1, setEnablePan1] = React.useState(true);
  const enablePan2 = useSharedValue(true);

  const pan1 = Gesture.Pan().enabled(enablePan1);

  const pan2 = usePan({
    enabled: enablePan2,
  });

  const as = useAnimatedStyle(() => {
    return {
      backgroundColor: enablePan2.value ? 'lightgreen' : 'crimson',
    };
  });

  return (
    <GestureHandlerRootView style={styles.container}>
      <Button
        title="Toggle old API enabled!"
        onPress={() => {
          setEnablePan1((prev) => !prev);
        }}
      />

      <GestureDetector gesture={pan1}>
        <View
          style={[
            styles.box,
            { backgroundColor: enablePan1 ? 'lightgreen' : 'crimson' },
          ]}
        />
      </GestureDetector>

      <Button
        title="Toggle new API enabled!"
        onPress={() => {
          enablePan2.value = !enablePan2.value;
        }}
      />

      <Button
        title="Toggle NativeDetetor"
        onPress={() => {
          setShowDetector((prev) => !prev);
        }}
      />

      {showDetector && (
        <NativeDetector gesture={pan2}>
          <Animated.View style={[styles.box, as]} />
        </NativeDetector>
      )}
    </GestureHandlerRootView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  box: {
    width: 200,
    height: 200,
    borderRadius: 20,
  },
});
```

</details>

m-bert commented on packages/react-native-gesture-handler/src/web/handlers/GestureHandler.ts:
```diff
@@ -707,16 +707,28 @@ export default abstract class GestureHandler implements IGestureHandler {
-      this._enabled = config.enabled;
-      this.delegate.onEnabledChange();
-    }
+    this.updateEnabled(config.enabled);
```
Let's break it down into two cases (note, I assume that default value for `this.enabled` is `true`):

1. `config.enabled === undefined`

In this case second `if` will always execute, regardless of `enabled` change. 

2. `config.enabled !== undefined`

Second `if` will never execute, as those values will be the same.

Or have I missed something?

akwasniewski replied:
Ok, you are right

m-bert commented on packages/react-native-gesture-handler/src/web/tools/GestureHandlerWebDelegate.ts:
```diff
@@ -48,17 +46,15 @@ export class GestureHandlerWebDelegate
       touchAction: this.view.style.touchAction,
     };
 
-    this.setUserSelect();
-    this.setTouchAction();
-    this.setContextMenu();
```
With changes in this PR, `onEnabledChange` is called so this is no longer necessary. This is because `setConfig` resets `enabled` into `null` and then in `updateEnabled` we call this method on delegate. It didn't work that way before though, so calls in `init` were necessary.

j-piasecki commented on packages/react-native-gesture-handler/src/web/handlers/GestureHandler.ts:
```diff
@@ -707,16 +707,28 @@ export default abstract class GestureHandler implements IGestureHandler {
+  }
+
   public setGestureConfig(config: Config) {
     this.resetConfig();
```
Can we reverse that and start with detached listeners which are then attached when config has `enabled !== false`?

m-bert replied:
So instead of attaching listeners in `resetConfig`, we should detach them, and then attach if `enabled !== false`?

j-piasecki replied:
I'm not sure I see it. Could you do a step-by-step on what needs to happen to get that?

j-piasecki replied:
Resolved offline - there's still an issue, no matter the starting value. Ternary logic is likely the simplest solution here.

m-bert commented:
I've changed it a bit and removed reset of `enabled` in `resetConfig` as discussed offline. I've also updated an example on which I've tested these changed. Let me know what you think (cc @j-piasecki)
